src/coreclr/jit/codegenarm64.cpp

```diff
@@ -2235,15 +2235,25 @@ void CodeGen::instGen_Set_Reg_To_Imm(emitAttr       size,
 {
     // reg cannot be a FP register
     assert(!genIsValidFloatReg(reg));
+
+    emitAttr origAttr = size;
     if (!compiler->opts.compReloc)
     {
         size = EA_SIZE(size); // Strip any Reloc flags from size if we aren't doing relocs
     }
 
     if (EA_IS_RELOC(size))
     {
-        // This emits a pair of adrp/add (two instructions) with fix-ups.
-        GetEmitter()->emitIns_R_AI(INS_adrp, size, reg, imm DEBUGARG(targetHandle) DEBUGARG(gtFlags));
+        if (!EA_IS_CNS_SEC_RELOC(origAttr))
+        {
+            // This emits a pair of adrp/add (two instructions) with fix-ups.
+            GetEmitter()->emitIns_R_AI(INS_adrp, size, reg, imm DEBUGARG(targetHandle) DEBUGARG(gtFlags));
+        }
+        else if (compiler->IsTargetAbi(CORINFO_NATIVEAOT_ABI))
+        {
+            // This emits pair of `add` instructions for TLS reloc
+            GetEmitter()->emitIns_Add_Add_Tls_Reloc(size, reg, imm DEBUGARG(gtFlags));
+        }
```

Comment:
```suggestion
        else
        {
            assert(compiler->IsTargetAbi(CORINFO_NATIVEAOT_ABI));
            // This emits pair of `add` instructions for TLS reloc
            GetEmitter()->emitIns_Add_Add_Tls_Reloc(size, reg, imm DEBUGARG(gtFlags));
        }
```
? Otherwise it looks like the function won't emit any codegen.

Reply:
We should not call `emitIns_Add_Add_Tls_Reloc` for JIT. The assert won't hit in Release build, so better to guard it with if-check.

Reply:
It does not look right to me to have a case where this function does not emit any instructions. If that happens it would result in silent bad codegen.

Reply:
Ah, yes. fixed.